src/main/java/randoop/generation/CoverageTracker.java

```diff
@@ -0,0 +1,199 @@
+   */
+  public void setClassesUnderTest(Set<String> classesUnderTest) {
+    this.classesUnderTest = new HashSet<>(classesUnderTest);
+  }
+
+  /**
+   * Attempts to instrument a class and return the byte array representation.
+   *
+   * @param className name of the class to instrument
+   * @return the instrumented bytes of the class, null if instrumentation failed
+   */
+  public byte[] instrumentClass(String className) {
+    // Only instrument classes under test and their nested classes.
+    if (!classesUnderTest.contains(className) && !isNestedClass(className)) {
+      return null;
+    }
+
+    final byte[] instrumented;
+    final String resource = '/' + className.replace('.', '/') + ".class";
+    InputStream original = getClass().getResourceAsStream(resource);
+
+    if (original == null) {
+      throw new Error("No resource with name: " + resource + " found by CoverageTracker!");
+    }
+
+    try {
+      // Instrument the class to prepare for coverage collection later.
+      instrumented = instrumenter.instrument(original, className);
+      original.close();
+    } catch (IOException e) {
+      e.printStackTrace();
+      return null;
+    }
+
+    return instrumented;
+  }
+
+  /**
+   * Determine if the given class is a nested class of a class under test.
+   *
+   * @param className the name of the class to check
+   * @return true iff the given class is a nested class of a class under test
+   */
+  private boolean isNestedClass(String className) {
+    for (String classUnderTest : classesUnderTest) {
+      if (className.startsWith(classUnderTest + "$")) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Returns the instrumented version of the class.
+   *
+   * @param className name of the class
+   * @return {@code Class} object that has been instrumented for coverage data collection. Returns
+   *     null if class with target name cannot be found.
+   */
+  public Class<?> getInstrumentedClass(String className) {
+    if (className == null) {
+      return null;
+    }
+
+    Class<?> instrumentedClass = null;
+    try {
+      instrumentedClass = memoryClassLoader.loadClass(className);
+    } catch (ClassNotFoundException e) {
+      e.printStackTrace();
+    }
+    return instrumentedClass;
+  }
+
+  /**
+   * Collect coverage information for all methods under test. At this point, coverage data has
+   * already been generated as Randoop has been constructing and executing its test sequences.
+   * Coverage data is now collected and summarized. The {@code coverageDetailsMap} is updated to
+   * contain the updated coverage information of each method branch.
+   */
+  public void collect() {
+    // Collect coverage information.
+    data.collect(executionData, sessionInfos, false);
+
+    CoverageBuilder coverageBuilder = new CoverageBuilder();
+    Analyzer analyzer = new Analyzer(executionData, coverageBuilder);
+
+    // Analyze the coverage of each of the tracked classes.
```

Comment:
This comment is not very descriptive.  It's the same as the method name, so it doesn't add much beyond that.  It would be better to say what the method actually does.